From e83e8290ec67b65ce2b25038b982cbe7c8e80c2e Mon Sep 17 00:00:00 2001
From: Shubhanshu Mani Tripathi <shubhanshu.mani.tripathi@emerson.com>
Date: Thu, 6 Nov 2025 12:02:36 +0530
Subject: [PATCH] onboard: fix compiler compatibilities issue

Bitbake build was failing with below error:

Onboard/osk/osk_module.h:28:16: error: cannot use keyword 'false'
as enumeration constant
   28 | typedef enum { false, true } bool;

Removed that typedef and used <stdbool.h> instead as 'bool enum'
was historically used to define a bool type in plain C
(before <stdbool.h> existed), but now it conflicts with standard
keywords.

Upstream-Status: Inactive-Upstream [lastcommit: 2017-10-09]

Signed-off-by: Shubhanshu Mani Tripathi <shubhanshu.mani.tripathi@emerson.com>
---
 Onboard/osk/osk_module.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/Onboard/osk/osk_module.h b/Onboard/osk/osk_module.h
index f611f7b..f520d60 100644
--- a/Onboard/osk/osk_module.h
+++ b/Onboard/osk/osk_module.h
@@ -24,8 +24,7 @@
 
 #include <Python.h>
 #include <structmember.h>
-
-typedef enum { false, true } bool;
+#include <stdbool.h>
 
 /**
  * Python2 to Python3 conversion
